diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index cb9cee378870..76868ac243db 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -16,6 +16,22 @@
 		base_dtb = "am335x-boneblack.dts";
 		base_dtb_timestamp = __TIMESTAMP__;
 	};
+
+	reserved-memory {
+        #address-cells = <1>;
+        #size-cells = <1>;
+        ranges;
+
+        ramoops@88000000 {
+            #address-cells = <1>;
+            #size-cells = <1>;
+            compatible = "ramoops";
+            reg = <0x88000000 0x100000>; // 1MB at 0x88000000
+            record-size = <0x10000>; // 64kB per record
+            console-size = <0x10000>; // 64kB for console output
+            pmsg-size = <0x1000>; // 4kB for user messages
+        };
+    };
 };
 
 &cpu0_opp_table {
